src/Stryker.Core/Stryker.Core/Options/StrykerInputs.cs

```diff
@@ -134,7 +135,7 @@ public StrykerOptions ValidateAll()
                 TestProjects = TestProjectsInput.Validate(),
                 TestCaseFilter = TestCaseFilterInput.Validate(),
                 DashboardUrl = DashboardUrlInput.Validate(),
-                DashboardApiKey = DashboardApiKeyInput.Validate(WithBaselineInput.SuppliedInput),
+                DashboardApiKey = DashboardApiKeyInput.Validate(reporters.Any(x => x == Reporters.Reporter.Dashboard) || (WithBaselineInput.SuppliedInput.GetValueOrDefault() && baselineProvider == Baseline.Providers.BaselineProvider.Dashboard)),
```

Comment:
I would prefer if the DashboardApiKeyInput does the filtering and simply receives the reporters and the baseline provider.